nntrainer/tensor/var_grad.cpp

```diff
@@ -20,7 +20,8 @@ Var_Grad::Var_Grad(const Var_Grad &rhs) :
   trainable(rhs.trainable),
   name(rhs.name) {
   var = rhs.var.clone();
-  grad = rhs.grad.clone();
+  if (rhs.trainable)
```

Comment:
How about,
```cpp
if (rhs.grad.uninitialized())
```

If it is not trainable but has a grad, it would mean something special we should care about.

if trainable means always gradient uninitialized, I think we wouldn't need trainable variable rather we can check if gradient is uninitialized to keep a single source of truth

Reply:
Added as a new commit (to make this update separate).